.github/workflows/docker.yml

```diff
@@ -0,0 +1,81 @@
+# The present workflow was made based on the following references:
+# - https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching
+# - https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md
+---
+name: Docker
+
+on: push
+
+concurrency:
+  group: docker-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/setup-buildx-action@v1
+      - uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-buildx-
+          path: /tmp/.buildx-cache
+      - uses: docker/build-push-action@v2
+        with:
+          context: .
+          tags: ${{ env.IMAGE_NAME }}
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,dest=/tmp/.buildx-cache-new,mode=max
+          load: true
+      - run: docker run ${{ env.IMAGE_NAME }} --version
+      - run: |
+          # this is a workaround, check the following issues:
+          # - https://github.com/docker/build-push-action/issues/252
+          # - https://github.com/moby/buildkit/issues/1896
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
+  publish:
+    if: contains(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    needs: [build]
```

Comment:
@sestrella Is it possible to add something like this for the `build` step but referring to `build` job from the `ci` workflow? In that way, if the CI workflow fails then this workflow won't run.